service/src/test/java/uk/gov/hmcts/reform/fpl/service/InboxLookupServiceTest.java

```diff
@@ -0,0 +1,95 @@
+import uk.gov.hmcts.reform.fpl.config.LocalAuthorityEmailLookupConfiguration;
+import uk.gov.hmcts.reform.fpl.model.Solicitor;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = {JacksonAutoConfiguration.class, LocalAuthorityEmailLookupConfiguration.class,
+    InboxLookupService.class, GeneralInboxLookupConfiguration.class})
+public class InboxLookupServiceTest {
+
+    @MockBean
+    private LocalAuthorityEmailLookupConfiguration localAuthorityEmailLookupConfiguration;
+
+    @MockBean
+    private GeneralInboxLookupConfiguration generalInboxLookupConfiguration;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    private InboxLookupService inboxLookupService;
+
+    private CaseDetails caseDetails;
+
+    private static final String LOCAL_AUTHORITY_CODE = "example";
+    private static final String LOCAL_AUTHORITY_EMAIL_ADDRESS = "FamilyPublicLaw+sa@gmail.com";
+    private static final String SOLICITOR_EMAIL_ADDRESS = "FamilyPublicLaw+sa@gmail.com";
+    private static final String GENERAL_INBOX = "FamilyPublicLaw+generalInbox@gmail.com";
+
+    @BeforeEach
+    void setup() {
+        this.inboxLookupService =
+            new InboxLookupService(mapper,
+                localAuthorityEmailLookupConfiguration,
+                generalInboxLookupConfiguration);
+    }
+
+    @Test
+    void shouldReturnLocalAuthorityInboxWhenMappingExist() {
+        buildCaseDetails(SOLICITOR_EMAIL_ADDRESS);
+        getMockLocalAuthorityEmail(LOCAL_AUTHORITY_EMAIL_ADDRESS);
+
+        String email = inboxLookupService.getLocalAuthorityOrFallbackEmail(caseDetails, LOCAL_AUTHORITY_CODE);
+
+        assertThat(email).isEqualTo(LOCAL_AUTHORITY_EMAIL_ADDRESS);
+    }
+
+    @Test
+    void shouldReturnSolicitorEmailWhenLocalAuthorityInboxNotExist() {
+        buildCaseDetails(SOLICITOR_EMAIL_ADDRESS);
+        getMockLocalAuthorityEmail("");
+
+        String email = inboxLookupService.getLocalAuthorityOrFallbackEmail(caseDetails, LOCAL_AUTHORITY_CODE);
+
+        assertThat(email).isEqualTo(SOLICITOR_EMAIL_ADDRESS);
+    }
+
+    @Test
+    void shouldReturnGeneralInboxWhenSolicitorEmailNotExist() {
+        buildCaseDetails("");
+        getMockLocalAuthorityEmail("");
+
+        given(generalInboxLookupConfiguration.getGeneralInbox())
+            .willReturn(GENERAL_INBOX);
+
+        String email = inboxLookupService.getLocalAuthorityOrFallbackEmail(caseDetails, LOCAL_AUTHORITY_CODE);
+
+        assertThat(email).isEqualTo(GENERAL_INBOX);
+    }
+
+    private void getMockLocalAuthorityEmail(String localAuthorityEmailAddress) {
+        given(localAuthorityEmailLookupConfiguration.getLocalAuthority(LOCAL_AUTHORITY_CODE))
+            .willReturn(new LocalAuthorityEmailLookupConfiguration.LocalAuthority(localAuthorityEmailAddress));
+    }
+
+    private void buildCaseDetails(String email) {
```

Comment:
Why not return casedetails?

Reply:
changed to return CaseDetails

Reply:
as suggested changed method name to `buildCaseDetailsWithSolicitorEmail`